File: share_folder/word_count/src/main/scala/WordCount.scala
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.SaveMode

object WordCount {

  def main(args: Array[String]) {

    //  Logger.getLogger("info").setLevel(Level.OFF)

    if (args.length != 2) {
      Console.err.println(s"Given ${args.length} arguments. ${args.toList}")
      Console.err.println("Need two arguments: <file in> <file out>")
      sys.exit(1)
    }

    val fIn = args(0)
    val fOut = args(1)

    val spark = SparkSession
      .builder()
      .appName("Word Count")
      .config("spark.some.config.option", "algun-valor")
      .getOrCreate()

    import spark.implicits._

    val sc = spark.sparkContext

    sc.setLogLevel("WARN")

    val lines = sc.textFile(fIn)

    println("Cantidad de tareas/particiones: " + lines.partitions.size)

    val words = lines
                .flatMap(l => l.split(" "))
                .filter(l => ! l.isEmpty)

    //MapReduce
    val wordCount = words.map(x => (x,1))
                    .reduceByKey((nx,ny) => nx+ny)

    // ordena por cantidad
    val result = wordCount.sortBy((p => p._2), ascending = false)

    result.toDS.write.mode(SaveMode.Overwrite).csv(fOut)
  }
}

File: share_folder/word_count/src/main/python/WordCount.py
import sys

from pyspark.sql import SparkSession

file_name = sys.argv[1]

spark = SparkSession.builder.appName("Word Count").getOrCreate()

sc = spark.sparkContext

lines = sc.textFile(file_name)

words = lines \
    .flatMap(lambda line: line.split(" ")) \
    .filter(lambda word: word)

#MapReduce
wordCount = words \
    .map(lambda word: (word,1)) \
    .reduceByKey(lambda n,m: n+m)

result = wordCount \
    .sortBy((lambda p: p[1]), ascending = False) # ordena por cantidad

local_result = result.collect() # Traigo desde cluster

for word, count in local_result[:10]: # tomo 10
    print(word, count) # los imprimo

spark.stop()
